From 2586e20fa7512ec8d3299bbd0a1c9f56b9c192ca Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Thu, 8 Feb 2024 19:12:38 +0100
Subject: [PATCH] CMakeLists.txt: fixing TMPDIR injections

Using cmake targets instead of INCLUDES leaking TMPDIR location to
cmake config files.

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeLists.txt | 29 ++++++++++-------------------
 1 file changed, 10 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47f22a9b6..d11c0f459 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,16 +18,6 @@ include(ConfigExtras.cmake)
 
 include_directories(include)
 
-set(THIS_PACKAGE_INCLUDE_DEPENDS
-  Boost
-  moveit_core
-  rclcpp
-  warehouse_ros
-  moveit_ros_planning
-  tf2_eigen
-  tf2_ros
-)
-
 # Libraries
 add_library(moveit_warehouse SHARED
   src/moveit_message_storage.cpp
@@ -42,7 +32,16 @@ include(GenerateExportHeader)
 generate_export_header(moveit_warehouse)
 target_include_directories(moveit_warehouse PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(moveit_warehouse PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_warehouse ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_warehouse PUBLIC
+  Boost::boost
+  Boost::program_options
+  moveit_core::moveit_utils
+  rclcpp::rclcpp
+  warehouse_ros::warehouse_ros
+  moveit_ros_planning::moveit_planning_scene_monitor
+  tf2_eigen::tf2_eigen
+  tf2_ros::tf2_ros
+)
 
 
 # Executables
@@ -94,14 +93,6 @@ install(
 
 ament_export_targets(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
 
-ament_export_dependencies(rclcpp)
-ament_export_dependencies(moveit_core)
-ament_export_dependencies(warehouse_ros)
-ament_export_dependencies(moveit_ros_planning)
-ament_export_dependencies(tf2_eigen)
-ament_export_dependencies(tf2_ros)
-ament_export_dependencies(Boost)
-
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
 
-- 
2.43.0

